pdns/auth-catalogzone.cc

```diff
@@ -108,6 +108,9 @@ std::string CatalogInfo::toJson() const
 void CatalogInfo::updateHash(CatalogHashMap& hashes, const DomainInfo& di) const
 {
   hashes[di.catalog].process(static_cast<char>(di.id) + di.zone.toLogString() + "\0" + d_coo.toLogString() + "\0" + d_unique.toLogString());
+  for (const auto& group : d_group) {
+    hashes[di.catalog].process(static_cast<char>(group.length()) + group);
```

Comment:
I find this a somewhat obscure construct. First I had to realize is `char + string` and not `char + char *`, which wold not work at all as expected. Why not use `std::to_string(group(length())`? Also, I do wonder why the length is explicitly hashed.
Similar issue with `static_cast<char>(di.id)` above. If `di.id` does not fit into a char you loose significant bits.

Reply:
Last comment: is it wise to create all the tmp strings? You could hash the individual parts separately, working with the raw presentation: `process(di.zone.getStorage())` etc...